FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -178,25 +208,46 @@ final class MessageService {
         )
         let messageType: ProcessedMessage.MessageType
         let text: String
+        let signature: ProcessedMessage.MessageSignature
 
         if let decryptErrBlock = decryptErrBlocks.first {
             let rawMsg = decryptErrBlock.content
             let err = decryptErrBlock.decryptErr?.error
             text = "Could not decrypt:\n\(err?.type.rawValue ?? "UNKNOWN"): \(err?.message ?? "??")\n\n\n\(rawMsg)"
             messageType = .error(err?.type ?? .other)
+            signature = .unknown
         } else {
             text = decrypted.text
             messageType = decrypted.replyType == CoreRes.ReplyType.encrypted ? .encrypted : .plain
+            signature = await verifySignature(longid: decrypted.blocks.first?.verifyRes?.signer,
+                                              email: sender)
         }
 
         return ProcessedMessage(
             rawMimeData: rawMimeData,
             text: text,
             attachments: attachments,
-            messageType: messageType
+            messageType: messageType,
+            signature: signature
         )
     }
 
+    private func verifySignature(longid: String?, email: String?) async -> ProcessedMessage.MessageSignature {
```

Comment:
let's call it `evaluateSignatureVerificationResult`

Also, instead of just `longid`, give it whole `VerifyRes?`.

Also edit `VerifyRes` type - it may have an `error` on it of type `String?`.

Here is equivalent method from the browser: https://github.com/FlowCrypt/flowcrypt-browser/blob/fc8db5bf7f05e6548459f463fe672c4e09c34e20/extension/chrome/elements/pgp_block_modules/pgp-block-signature-module.ts#L24

Below is pseudo code for you in the context of this swift app.

```ts
  private func renderPgpSignatureCheckResult = (
    signature: VerifyRes?, 
    sender: RecipientWithOrderedPublicKeys
  ) -> ProcessedMessage.MessageSignature {
    if signature is nil then return .unsigned // render "Message not signed" in red
    if signature.error is not nil then return .error(signature.error) // render the error in red
    if signature.signer is nil then return .unsigned // render "Message not signed" in red
    if signature.match is nil then return .unsigned // render "Message not signed" in red
    if signature.match is false then return .bad // render "Bad signature" in red
    let signingPubKey = sender.pubkeys.find { $0.longids.contains(signature.signer) }
    if signingPubKey is nil then return .missingPubkey( 
       longid: signature.signer, 
       senderEmail: sender.email
     ) // render "Missing public key \(longid) to verify signature from \(senderEmail)" in red
     return .good
  }
```

Reply:
Thanks for detailed explanation 👍
I checked this browser code during development too, to see how it renders signature block, it was very helpful.